src/main/java/io/github/cottonmc/resources/oregen/CottonOreFeature.java

```diff
@@ -1,138 +1,110 @@
+					
+					clusterX += toGenerateIn.getPos().getStartX();
+					clusterZ += toGenerateIn.getPos().getStartZ();
+					
+					int generatedThisCluster = generateVeinPart(world, clusterX, clusterY, clusterZ, clusterSize, radius, settings.ores, 85, rand);
+					blocksGenerated += generatedThisCluster;
 				}
+				
+				//System.out.println("    Generated "+blocksGenerated+" in "+clusters+" clusters.");
+			} else {
+				//System.out.println("    skipping "+s+" here.");
 			}
 		}
-
+		
 		return false;
 	}
-
-	protected boolean generateVeinPart(IWorld world, Random rand, CottonOreFeatureConfig config, double double_1, double double_2, double double_3, double double_4, double double_5, double double_6, int int_1, int int_2, int int_3, int int_4, int int_5) {
-		int int_6 = 0;
-		BitSet bitSet_1 = new BitSet(int_4 * int_5 * int_4);
-		BlockPos.Mutable blockPos$Mutable_1 = new BlockPos.Mutable();
-		double[] doubles_1 = new double[config.size * 4];
-
-		int int_8;
-		double double_12;
-		double double_13;
-		double double_14;
-		double double_15;
-		for(int_8 = 0; int_8 < config.size; ++int_8) {
-			float float_1 = (float)int_8 / (float)config.size;
-			double_12 = MathHelper.lerp((double)float_1, double_1, double_2);
-			double_13 = MathHelper.lerp((double)float_1, double_5, double_6);
-			double_14 = MathHelper.lerp((double)float_1, double_3, double_4);
-			double_15 = rand.nextDouble() * (double)config.size / 16.0D;
-			double double_11 = ((double)(MathHelper.sin(3.1415927F * float_1) + 1.0F) * double_15 + 1.0D) / 2.0D;
-			doubles_1[int_8 * 4 + 0] = double_12;
-			doubles_1[int_8 * 4 + 1] = double_13;
-			doubles_1[int_8 * 4 + 2] = double_14;
-			doubles_1[int_8 * 4 + 3] = double_11;
-		}
-
-		for(int_8 = 0; int_8 < config.size - 1; ++int_8) {
-			if (doubles_1[int_8 * 4 + 3] > 0.0D) {
-				for(int int_9 = int_8 + 1; int_9 < config.size; ++int_9) {
-					if (doubles_1[int_9 * 4 + 3] > 0.0D) {
-						double_12 = doubles_1[int_8 * 4 + 0] - doubles_1[int_9 * 4 + 0];
-						double_13 = doubles_1[int_8 * 4 + 1] - doubles_1[int_9 * 4 + 1];
-						double_14 = doubles_1[int_8 * 4 + 2] - doubles_1[int_9 * 4 + 2];
-						double_15 = doubles_1[int_8 * 4 + 3] - doubles_1[int_9 * 4 + 3];
-						if (double_15 * double_15 > double_12 * double_12 + double_13 * double_13 + double_14 * double_14) {
-							if (double_15 > 0.0D) {
-								doubles_1[int_9 * 4 + 3] = -1.0D;
-							} else {
-								doubles_1[int_8 * 4 + 3] = -1.0D;
-							}
-						}
-					}
+	
+	protected int generateVeinPart(IWorld world, int x, int y, int z, int clumpSize, int radius, Set<BlockState> states, int density, Random rand) {
+		int rad2 = radius * radius;
+		BlockState[] blocks = states.toArray(new BlockState[states.size()]);
+		int replaced = 0;
+		
+		//TODO: Maybe add these into a set and scramble them so that ores distribute more evenly into the sphere
+		//      Alternatively, we could spray ores *at random* using a gaussian distribution. Not sure which is better.
```

Comment:
I think a gaussian distribution would be nice here tbh. It'd feel a bit more natural.

Reply:
Taking a list and randomizing its order would cause uniform distribution throughout the sphere. Randomly spraying ores would create a normal (gaussian) distribution with more ores at the center than the edges. The list approach is also more deterministic as to how many ores are created; we might have to do more spraying to approach a particular block quota.

I'm still leaning more towards gaussian than list because randomizing a list is O(n) and we might come in significantly under that by random chance.

Reply:
Heck, I managed to combine them! This winds up being really efficient, and the ores look more cohesive because I'm picking the right radius for a given number of blocks, and preferring blocks central to that clump.

Reply:
nice!